app/controllers/metadata_controller.rb

```diff
@@ -124,6 +124,21 @@ def default_manifest
       manifest[:icons] << icon_entry
     end
 
+    SiteSetting.manifest_screenshots.split('|').each do |image|
+      next unless Discourse.store.has_been_uploaded?(image)
+
+      upload = Upload.find_by(url: image.delete_prefix(Discourse.base_url))
```

Comment:
Does this work for S3 uploads?

Reply:
Oh yeah, that will fail.

I tried to find a method to go from Upload full URL to Upload object but did not find a ready made method for that, are you aware of any?

Reply:
Maybe extract the SHA and then lookup based on that?

https://github.com/discourse/discourse/blob/a96a5db0fb516c3b88b225827eda47af63995d64/app/models/post.rb#L1013-L1018

Reply:
Should work now @davidtaylorhq !